public/app/plugins/datasource/grafana/datasource.ts

```diff
@@ -6,54 +6,63 @@ import {
   DataQueryResponse,
   DataSourceApi,
   DataSourceInstanceSettings,
+  LiveChannelScope,
 } from '@grafana/data';
 
-import { GrafanaQuery, GrafanaAnnotationQuery, GrafanaAnnotationType } from './types';
-import { getBackendSrv, getTemplateSrv, TemplateSrv, toDataQueryResponse } from '@grafana/runtime';
+import { GrafanaQuery, GrafanaAnnotationQuery, GrafanaAnnotationType, GrafanaQueryType } from './types';
+import { getBackendSrv, getTemplateSrv, toDataQueryResponse, getLiveMeasurementsObserver } from '@grafana/runtime';
 import { Observable, of } from 'rxjs';
 import { map, catchError } from 'rxjs/operators';
 
+let counter = 100;
+
 export class GrafanaDatasource extends DataSourceApi<GrafanaQuery> {
-  constructor(
-    instanceSettings: DataSourceInstanceSettings,
-    private readonly templateSrv: TemplateSrv = getTemplateSrv()
-  ) {
+  constructor(instanceSettings: DataSourceInstanceSettings) {
     super(instanceSettings);
   }
 
   query(request: DataQueryRequest<GrafanaQuery>): Observable<DataQueryResponse> {
-    const { intervalMs, maxDataPoints, range, requestId } = request;
-
-    // Yes, this implementaiton ignores multiple targets!  But that matches exisitng behavior
-    const params: Record<string, any> = {
-      intervalMs,
-      maxDataPoints,
-      from: range.from.valueOf(),
-      to: range.to.valueOf(),
-    };
-
-    return getBackendSrv()
-      .fetch({
-        url: '/api/tsdb/testdata/random-walk',
-        method: 'GET',
-        params,
-        requestId,
-      })
-      .pipe(
-        map((rsp: any) => {
-          return toDataQueryResponse(rsp);
-        }),
-        catchError(err => {
-          return of(toDataQueryResponse(err));
-        })
-      );
+    const queries: Array<Observable<DataQueryResponse>> = [];
+    for (const target of request.targets) {
+      if (target.hide) {
+        continue;
+      }
+      if (target.queryType === GrafanaQueryType.LiveMeasurements) {
+        const { channel, measurements } = target;
+        if (channel) {
+          queries.push(
+            getLiveMeasurementsObserver(
+              {
+                scope: LiveChannelScope.Grafana,
+                namespace: 'measurements',
+                path: channel,
+              },
+              `${request.requestId}.${counter++}`,
+              measurements
+            )
+          );
+        }
+      } else {
+        queries.push(getRandomWalk(request));
+      }
+    }
+    // With a single query just return the results
+    if (queries.length === 1) {
+      return queries[0];
+    }
+    if (queries.length > 1) {
+      // HELP!!!
+      return queries[0];
+    }
+    return of(); // nothing
```

Comment:
not sure I understand this part, and since we are returning getLiveMeasurementsObserver and it calls getStream on the stream handler, when this is unsubscribed by Grafana why is not the centrifuge channel subscription also then removed?